### ISSUE TITLE
Null reference exception when multi-selecting array properties where one value is null

### ISSUE THREAD

**Issue by glopesdev** [bonsai-rx/bonsai#862]
For example, in the following workflow where one of the `RoiActivity` nodes has a defined `Regions` property and the second one has the default `null` value:

![image](https://user-images.githubusercontent.com/5315880/158352515-25add9c0-39fc-48e2-8fde-e38d707bd797.png)

When dragging the mouse cursor across both nodes to define a multi-selection, the property grid will attempt to merge both properties and currently throws a null reference exception. If these could be avoided, that would be preferrable.

**Comment by RoboDoig** [comment]
This can technically be fixed by having RoiActivity Regions initialized to an empty array rather than null.

The root issue is that on a multi-selection, PropertyGridInternal is trying to merge collections of selected objects. If the elements of this collection have properties that are null, the merge will error out and lead to this bug.

Would this work for all modules though? In some cases null conceptually means something different than e.g. empty array initialization.

**Comment by glopesdev** [comment]
@RoboDoig indeed, this seems to be a bug in `PropertyGridInternal`, specifically in [`MergePropertyDescriptor.MergeCollection`](https://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/WinForms/PropertyGridInternal/MergePropertyDescriptor.cs,494) and [`MergePropertyDescriptor.GetValue`](https://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/WinForms/PropertyGridInternal/MergePropertyDescriptor.cs,280) methods where no null check is performed when merging collections.

There is little which can be done at the moment, since the goal was to resolve this for the general case, and as you mention it is often the case that null is indicative of a special case value.

Let's leave the issue here for now, but remove the milestone 2.7.0 and we can review it later more thoroughly while evaluating the 3.0 release.
